services/src/modules/resource-repository/fs.ts

```diff
@@ -26,12 +28,30 @@ export class FileSystemResourceRepository implements ResourceRepository {
         await fs.writeFile(this.pathToFile, JSON.stringify(rg));
     }
 
+    async writePolicy(filename: string, compiledRego: Buffer): Promise<void> {
```

Comment:
I think that compileRego adds a bit of dependency, maybe it should abstract as a regular resource and add attachment support? (maybe with path conventions, or getAttachment or something)

Reply:
I agree about `compiledRego`, it was an oversight on my part, I will rename it to something more generic.

Regarding the policy, it is already a regular resource that is saved in the resources file as part of the resource group. This function only writes the attachment (compiled rego in the case of opa) and is generic, so it can work as attachment for other types as well (poor variable naming aside).
After explaining it now I realize the function name itself (`writePolicy`) is also bad since it doesn't write the policy itself (which is written with the regular `update` function) but only the attachment. I will rename the function itself as well.

Do you still think other changes are required here other than renaming the `compiledRego` variable to something like `content` and the `writePolicy` function to something like `writePolicyAttachment`?

As a note, the attachment itself is stored with a name convention based on the metadata of namespace and name, there is no explicit link to it in the policy resource

Reply:
@Yshayy This is addressed in commit 738408c